In [1]:
# Load numpy
import numpy as np

# Hypothesis:
###  The older the male patient is, as well as higher cholesterol levels, can increase the patient's likelyhood of getting or having heart disease. 

In [2]:
# Load the dataset
data=np.genfromtxt("heart.csv", delimiter=",", skip_header=True)

# 0 = male
# 1 = female

# 0 = no heart disease
# 1 = heart disease

In [3]:
# Columns of choice
age=data[:,0]
print(f"Age: {age[:5]}")
sex=data[:,1]
print(f"Sex: {sex[:5]}")
chol=data[:,4]
print(f"Cholosteral: {chol[:5]}")
trestbps=data[:,3]
print(f"Resting heartrate: {trestbps[:5]}")
thalach=data[:,7]
print(f"Maximum heart rate achieved: {thalach[:5]}")
target=data[:,-1]
print(f"Target: {target[:5]}")

Age: [63. 37. 41. 56. 57.]
Sex: [1. 1. 0. 1. 0.]
Cholosteral: [233. 250. 204. 236. 354.]
Resting heartrate: [145. 130. 130. 120. 120.]
Maximum heart rate achieved: [150. 187. 172. 178. 163.]
Target: [1. 1. 1. 1. 1.]


# Basic Descriptive Statistics
### Mean, Median, SD, Min, Max

In [4]:
#age
print("---Basic Descriptive Statistics on Age---")
print(f"Mean: {round(np.mean(age))}")
print(f"Median: {np.median(age)}")
print(f"Standard Deviation: {round(np.std(age))}")
print(f"Minimum: {np.min(age)}")
print(f"Maximum: {np.max(age)}")

---Basic Descriptive Statistics on Age---
Mean: 54
Median: 55.0
Standard Deviation: 9
Minimum: 29.0
Maximum: 77.0


In [5]:
#sex
print("---Basic Descriptive Statistics on Sex---")
print(f"Mean: {round(np.mean(sex))}")
print(f"Median: {np.median(sex)}")
print(f"Standard Deviation: {round(np.std(sex))}")
print(f"Minimum: {np.min(sex)}")
print(f"Maximum: {np.max(sex)}")

---Basic Descriptive Statistics on Sex---
Mean: 1
Median: 1.0
Standard Deviation: 0
Minimum: 0.0
Maximum: 1.0


In [6]:
#cholesteral
print("---Basic Descriptive Statistics on Cholesteral---")
print(f"Mean: {round(np.mean(chol))}")
print(f"Median: {np.median(chol)}")
print(f"Standard Deviation: {round(np.std(chol))}")
print(f"Minimum: {np.min(chol)}")
print(f"Maximum: {np.max(chol)}")

---Basic Descriptive Statistics on Cholesteral---
Mean: 246
Median: 240.0
Standard Deviation: 52
Minimum: 126.0
Maximum: 564.0


In [7]:
#resting heartrate
print("---Basic Descriptive Statistics on Resting Heartrate---")
print(f"Mean: {round(np.mean(trestbps))}")
print(f"Median: {np.median(trestbps)}")
print(f"Standard Deviation: {round(np.std(trestbps))}")
print(f"Minimum: {np.min(trestbps)}")
print(f"Maximum: {np.max(trestbps)}")

---Basic Descriptive Statistics on Resting Heartrate---
Mean: 132
Median: 130.0
Standard Deviation: 18
Minimum: 94.0
Maximum: 200.0


In [8]:
#maximum heartrate achieved
print("---Basic Descriptive Statistics on Maximum Heartrate Achieved---")
print(f"Mean: {round(np.mean(thalach))}")
print(f"Median: {np.median(thalach)}")
print(f"Standard Deviation: {round(np.std(thalach))}")
print(f"Minimum: {np.min(thalach)}")
print(f"Maximum: {np.max(thalach)}")

---Basic Descriptive Statistics on Maximum Heartrate Achieved---
Mean: 150
Median: 153.0
Standard Deviation: 23
Minimum: 71.0
Maximum: 202.0


In [9]:
#target
print("---Basic Descriptive Statistics on Target---")
print(f"Mean: {round(np.mean(target))}")
print(f"Median: {np.median(target)}")
print(f"Standard Deviation: {round(np.std(target))}")
print(f"Minimum: {np.min(target)}")
print(f"Maximum: {np.max(target)}")

---Basic Descriptive Statistics on Target---
Mean: 1
Median: 1.0
Standard Deviation: 0
Minimum: 0.0
Maximum: 1.0


# Filtering the Data

In [10]:
print(f"Number of rows before filtering: {len(data)}")

Number of rows before filtering: 303


In [25]:
# Filtering data between older male patients with a high cholesteral
filtered_data=data[(age>50) & (sex==0) & (chol>250) & (target==1)]
print(f"Number of rows after filtering: {len(filtered_data)}")

Number of rows after filtering: 28


In [26]:
# Filtering data between older male patients with a high cholesteral
filtered_data=data[(age>50) & (sex==0) & (chol>250) & (target==0)]
print(f"Number of rows after filtering: {len(filtered_data)}")

Number of rows after filtering: 13


In [27]:
# Filtering data between older female patients with a high cholesteral
filtered_data=data[(age>50) & (sex==1) & (chol>250) & (target==1)]
print(f"Number of rows after filtering: {len(filtered_data)}")

Number of rows after filtering: 12


In [28]:
# Filtering data between older female patients with a high cholesteral
filtered_data=data[(age>50) & (sex==1) & (chol>250) & (target==0)]
print(f"Number of rows after filtering: {len(filtered_data)}")

Number of rows after filtering: 45


# Number of Unique Categories

In [12]:
print(f"Unique age count: {np.unique(age)}")
print(f"How many unique ages: {len(np.unique(age))}")

Unique age count: [29. 34. 35. 37. 38. 39. 40. 41. 42. 43. 44. 45. 46. 47. 48. 49. 50. 51.
 52. 53. 54. 55. 56. 57. 58. 59. 60. 61. 62. 63. 64. 65. 66. 67. 68. 69.
 70. 71. 74. 76. 77.]
How many unique ages: 41


In [14]:
print(f"Unique cholesteral count: {np.unique(chol)}")
print(f"How many unique cholesteral: {len(np.unique(chol))}")

Unique cholesteral count: [126. 131. 141. 149. 157. 160. 164. 166. 167. 168. 169. 172. 174. 175.
 176. 177. 178. 180. 182. 183. 184. 185. 186. 187. 188. 192. 193. 195.
 196. 197. 198. 199. 200. 201. 203. 204. 205. 206. 207. 208. 209. 210.
 211. 212. 213. 214. 215. 216. 217. 218. 219. 220. 221. 222. 223. 224.
 225. 226. 227. 228. 229. 230. 231. 232. 233. 234. 235. 236. 237. 239.
 240. 241. 242. 243. 244. 245. 246. 247. 248. 249. 250. 252. 253. 254.
 255. 256. 257. 258. 259. 260. 261. 262. 263. 264. 265. 266. 267. 268.
 269. 270. 271. 273. 274. 275. 276. 277. 278. 281. 282. 283. 284. 286.
 288. 289. 290. 293. 294. 295. 298. 299. 300. 302. 303. 304. 305. 306.
 307. 308. 309. 311. 313. 315. 318. 319. 321. 322. 325. 326. 327. 330.
 335. 340. 341. 342. 353. 354. 360. 394. 407. 409. 417. 564.]
How many unique cholesteral: 152


In [15]:
print(f"Unique resting heartrate count: {np.unique(trestbps)}")
print(f"How many unique resting heartrate: {len(np.unique(trestbps))}")

Unique resting heartrate count: [ 94. 100. 101. 102. 104. 105. 106. 108. 110. 112. 114. 115. 117. 118.
 120. 122. 123. 124. 125. 126. 128. 129. 130. 132. 134. 135. 136. 138.
 140. 142. 144. 145. 146. 148. 150. 152. 154. 155. 156. 160. 164. 165.
 170. 172. 174. 178. 180. 192. 200.]
How many unique resting heartrate: 49


In [16]:
print(f"Unique maximum heartrate count: {np.unique(thalach)}")
print(f"How many unique maximum heartrates: {len(np.unique(thalach))}")

Unique maximum heartrate count: [ 71.  88.  90.  95.  96.  97.  99. 103. 105. 106. 108. 109. 111. 112.
 113. 114. 115. 116. 117. 118. 120. 121. 122. 123. 124. 125. 126. 127.
 128. 129. 130. 131. 132. 133. 134. 136. 137. 138. 139. 140. 141. 142.
 143. 144. 145. 146. 147. 148. 149. 150. 151. 152. 153. 154. 155. 156.
 157. 158. 159. 160. 161. 162. 163. 164. 165. 166. 167. 168. 169. 170.
 171. 172. 173. 174. 175. 177. 178. 179. 180. 181. 182. 184. 185. 186.
 187. 188. 190. 192. 194. 195. 202.]
How many unique maximum heartrates: 91


# Correlation

In [22]:
corr_heart=np.corrcoef(age, chol, rowvar=False)
print(f"Corr age and chol {corr_heart}")

Corr age and chol [[1.         0.21367796]
 [0.21367796 1.        ]]


In [23]:
corr_heart=np.corrcoef(trestbps, chol, rowvar=False)
print(f"Corr resting heartrate and chol {corr_heart}")

Corr resting heartrate and chol [[1.         0.12317421]
 [0.12317421 1.        ]]


In [24]:
corr_heart=np.corrcoef(thalach, chol, rowvar=False)
print(f"Corr maximum heartrate and chol {corr_heart}")

Corr maximum heartrate and chol [[ 1.         -0.00993984]
 [-0.00993984  1.        ]]


# Correlation Analysis
- Age and Cholesteral Levels: 21% (weak positive)
- Resting Heartrate and Cholesteral Levels: 12% (weak postive)
- Maximum Heartrate and Cholesteral Levels: -0.9% (weak negative)

# Summary

### In our hypothesis, we stated that the older a male patient is and the higher their cholesteral levels are, the more likely that a patients is to get heart disease. After going through and testing our data, we discovered that this was not the case and there is actually a completely different correlation between people getting heart disease. However, after we analyzed our data, we discovered more of a correlation between the male patients that are older and have a higher cholesteral levels having heart disease, compared to the older female patients with higher cholesteral levels.